### PR TITLE
busybox: add getedid script

### PR DESCRIPTION
**Pull request by awiouy**
davu can test for intel
needs to be tested for nvidia

### CODE REVIEW

File: packages/sysutils/busybox/scripts/getedid [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+# exit 6 = xorg.conf already exists in /storage/.config
+# exit 7 = no xorg.conf in /storage/.config
+
+
+
+# Help message and usage explanation
+help() {
+echo "This script generates a custom EDID depending on your GPU"
+echo ""
+echo "To check which GPU you are using, use: ./getedid.sh gpu"
```
Comment (chewitt) [inline packages/sysutils/busybox/scripts/getedid]: if the script is `getedid` the references to `getedid.sh` are incorrect

Comment (DaVukovic) [inline packages/sysutils/busybox/scripts/getedid]: The script should be getedid.sh

File: packages/sysutils/busybox/scripts/getedid.sh [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+
+# run this first if the user already has a custom EDID but want to create a new one (TV or AVR change)
+del_edid() {
+  if [ "$gpu" = "intel" ]; then
+    check_file
+    if [ -f "$file".old ];  then
+      mount_rw
+      rm "$file"
+      mv "$file".old "$file"
+      mount_ro
```
Comment (MilhouseVH) [inline packages/sysutils/busybox/scripts/getedid.sh]: Where is this?

File: packages/sysutils/busybox/scripts/getedid.sh [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+      rm -r /storage/cpio
+      rm -r /storage/.config/firmware
```
Comment (MilhouseVH) [inline packages/sysutils/busybox/scripts/getedid.sh]: I'd suggest `rm -fr` instead of `rm -r`

File: packages/sysutils/busybox/scripts/getedid.sh [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+
+# mounting /flash to rw
+mount_rw() {
+  mount -o remout,rw /flash
+}
+
+
+# remount /flash to ro and reboot
+sys_reboot() {
+  mount -o remount,ro /flash
```
Comment (MilhouseVH) [inline packages/sysutils/busybox/scripts/getedid.sh]: Use `mount_ro` once you've add it.

File: packages/sysutils/busybox/scripts/getedid.sh [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+
+# restore the backup done by the script and depending on the GPU
+restore() {
+
+  # intel
+  if [ "$gpu" = "intel" ]; then
+    check_file
+    mount_rw
+    if [ -f "$file".old ]; then
+      rm "$file"
```
Comment (MilhouseVH) [inline packages/sysutils/busybox/scripts/getedid.sh]: Not required as you overwrite it in the next line

File: packages/sysutils/busybox/scripts/getedid.sh [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+      rm -r /storage/cpio
+      rm -r /storage/.config/firmware
+      sys_reboot
+    else
+      echo "You don't have a backup file for $file. You didn't used this script to create the custom EDID"
+      echo "Therefore we can't be sure the script working prooerly. Exiting"
+      exit 3
+    fi
+  elif [ "$gpu" = "nvidia" ]; then
+    if [ -f /storage/.config/xorg.conf ]; then
```
Comment (MilhouseVH) [inline packages/sysutils/busybox/scripts/getedid.sh]: No need to check if it exists, just use `rm -f`

File: packages/sysutils/busybox/scripts/getedid.sh [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+  #check which output is connnected:
+  for i in /sys/class/drm/*; do
+    if [ "$(cat "$i"/status 2>/dev/null)" = "connected" ]; then
+      hdmi="$(echo "$i" | cut -d / -f 5 | sed 's/card[0-9]-//g')"
+    fi
+  done
+
+  #create edid
+  mkdir -p /storage/.config/firmware/edid
+  cat /sys/class/drm/"$hdmi"/edid > /storage/.config/firmware/edid/edid.bin
```
Comment (MilhouseVH) [inline packages/sysutils/busybox/scripts/getedid.sh]: `"/sys/class/drm/$hdmi/edid"`

**Pull request closed without merging**